src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.ts

```diff
@@ -204,59 +208,107 @@ export class TableInteractorColumnComponent implements OnChanges {
+              // The subcomponent of this interactor is part of the complex, we update the start and end indices for the subcomponents
+              // line as it may start in this subcomponent
+              enrichedComplex.startSubComponentIndex = this.getMinValue(enrichedComplex.startSubComponentIndex, k);
+              enrichedComplex.endSubComponentIndex = this.getMaxValue(enrichedComplex.endSubComponentIndex, k);
             }
+          }
         }
       }
     }
     return enrichedComplex;
   }
 
   public classifyInteractorsByOrganism() {
-    this.enrichedInteractors.sort((a, b) => b.interactor.organismName.localeCompare(a.interactor.organismName));
+    this.enrichedInteractors.sort((a, b) => {
+      if (b.interactor.organismName === a.interactor.organismName) {
+        return b.timesAppearing - a.timesAppearing;
+      } else {
+        const organismBTimesAppearing = this._timesAppearingByOrganism.get(b.interactor.organismName);
+        const organismATimesAppearing = this._timesAppearingByOrganism.get(a.interactor.organismName);
+        if (organismBTimesAppearing === organismATimesAppearing) {
+          return b.interactor.organismName.localeCompare(a.interactor.organismName);
+        } else {
+          return organismBTimesAppearing - organismATimesAppearing;
+        }
+      }
+    });
     this.rangeOfInteractorOrganism();
   }
 
   public classifyInteractorsByType() {
-    this.enrichedInteractors.sort((a, b) => b.interactor.interactorType.localeCompare(a.interactor.interactorType));
+    this.enrichedInteractors.sort((a, b) => {
+      if (b.interactor.interactorType === a.interactor.interactorType) {
+        return b.timesAppearing - a.timesAppearing;
+      } else {
+        const typeBTimesAppearing = this._timesAppearingByType.get(b.interactor.interactorType);
+        const typeATimesAppearing = this._timesAppearingByType.get(a.interactor.interactorType);
+        if (typeBTimesAppearing === typeATimesAppearing) {
+          return b.interactor.interactorType.localeCompare(a.interactor.interactorType);
+        } else {
+          return typeBTimesAppearing - typeATimesAppearing;
+        }
+      }
+    });
     this.rangeOfInteractorType();
   }
 
   public classifyInteractorsByOccurrence() {
+    this.enrichedInteractors.sort((a, b) =>
+      b.timesAppearing - a.timesAppearing
+    );
+    this.ranges = [];
+  }
+
+  private calculateTimesAppearingType() {
+    this._timesAppearingByType = new Map();
     for (const oneInteractor of this.enrichedInteractors) {
       for (const complex of this.complexes) {
         const match = findInteractorInComplex(complex, oneInteractor.interactor.identifier, this.enrichedInteractors);
         if (!!match) {
-          if (!!match.stochiometryValue) {
-            oneInteractor.timesAppearing += match.stochiometryValue[0];
+          oneInteractor.timesAppearing += 1;
```

Comment:
`oneInteractor.timesAppearing += 1;` is called in this method and the method below, so we're setting `timesAppearing` to double the real value.
 It doesn't matter much, as all interactors have their number duplicated, so they are still in order, but to fix it we have 2 options.

1. Merge this 2 methods into 1, that does it all. They both have the same for loops, so it will just moving the if/else from one method to another.

2. Keep these 2 methods, without the `oneInteractor.timesAppearing += 1;` line, and have another method, `calculateTimesAppearing()` that only does that. Then call the 3 method on the `onChanges` method.